src/appfl/comm/globus_compute/utils/s3_storage.py

```diff
@@ -165,41 +166,57 @@ def download_file(self, file_path, object_name = None, object_url = None, delete
             object_url: presigned url for downloading the file
             delete_cloud: delete the cloud object after downloading using object name (only possible with credentials)
         """
+        REPEAT_TIMES = 5
         if object_url is not None:
-            try:
-                response = requests.get(object_url)
-                if response.status_code == 200:
-                    with open(file_path, "wb") as file:
-                        file.write(response.content)
-                    if self.logger is not None:
-                        self.logger.info(f"Successfully donwload object using presigned url")
-                    else:
-                        print(f"Successfully donwload object using presigned url")
-                return
-            except:
-                if self.logger is not None:
-                    self.logger.info(f"Error in downloading object using presigned url")
-                else:
-                    print(f"Error in downloading object using presigned url")
-                raise
+            for i in range(REPEAT_TIMES):
+                try:
+                    response = requests.get(object_url)
+                    if response.status_code == 200:
+                        with open(file_path, "wb") as file:
+                            file.write(response.content)
+                        if self.logger is not None:
+                            self.logger.info(f"Successfully donwload object using presigned url")
+                        else:
```

Comment:
There is a typo in the word "download"

Reply:
Fixed the typo. Though using formatted string, let's just leave it as it is without additional parameters.